src/esp/bindings_js/CMakeLists.txt

```diff
@@ -33,3 +28,8 @@ foreach(resource ${resources})
       COMMAND ${CMAKE_COMMAND} -E create_symlink
           ${path} ${CMAKE_CURRENT_BINARY_DIR}/${filename})
 endforeach()
+
+set(build_command npm run build)
+message(STATUS "Building JS files now")
+execute_process(COMMAND ${build_command}
+  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
```

Comment:
To support incremental build, you could change this to:

```
set(bundle_files
    index.js
    navigate.js
    simenv_embind.js)

add_custom_command(OUTPUT bundle.js
                   COMMAND npm run build
                   DEPENDS hsim_bindings ${bundle_files}
                   WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
                   COMMENT "Building JS files now")

add_custom_target(hsim_bundle ALL
                  DEPENDS bundle.js)

add_dependencies(hsim_bundle hsim_bindings)

```

Reply:
Couldn't it be just 

```cmake
set(bundle_files
    index.js
    navigate.js
    simenv_embind.js)

add_custom_target(hsim_bundle ALL
                  COMMAND npm run build
                  DEPENDS hsim_bindings ${bundle_files}
                  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
                  COMMENT "Building JS files now")
```

instead? (From the top of my head, apologies if it doesn't work that way.)

Reply:
I think that will cause "npm run build" to always be run. At least, that has been my experience.

According to the docs, "The target has no output file and is always considered out of date".

Reply:
Right, true, forgot about that. Sorry about the noise then :)